### PR TITLE
firmware-qcom-sa8155p: Package SA8155p ADP firmware files

### PR DESCRIPTION
**Pull request by bhupesh-sharma**
Add recipe extracting SA8155p ADP firmware files from
Qualcomm's binary releases and packaging them into proper
binary package.

Signed-off-by: Bhupesh Sharma <bhupesh.sharma@linaro.org>

### CODE REVIEW

File: recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        install -d ${D}${sysconfdir}/
+        install -m 0644 LICENSE.qcom.txt ${D}${sysconfdir}/QCOM-LINUX-BOARD-SUPPORT-LICENSE-${PN}
+}
+
+FILES_${PN} += "${nonarch_base_libdir}/firmware/*"
+INSANE_SKIP_${PN} += "arch"
+INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
+INHIBIT_PACKAGE_STRIP = "1"
+INHIBIT_DEFAULT_DEPS = "1"
```
Comment (lumag) [inline recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb:38]: Please use `firmware-qcom.inc`

**Review by lumag — CHANGES_REQUESTED** [review]

File: recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+SA8155P_ADP_FW_NAME = "sa8155p_adp_fw"
+SA8155P_ADP_FW_ARCHIVE = "sa8155p_adp_fw.zip"
+
+SRC_URI_FW = "${SA8155P_ADP_FW_URI}${SA8155P_ADP_FW_ARCHIVE}"
+
+require recipes-bsp/firmware/firmware-qcom.inc
+DEPENDS += "qca-swiss-army-knife-native"
+
+COMPATIBLE_MACHINE = "(sa8155p)"
```
[the remark this answers is not in the document]
Comment (bhupesh-sharma) [reply recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb:19]: Ok.

File: recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+S = "${WORKDIR}/${SA8155P_ADP_FW_NAME}"
+
+do_install() {
+	cd ${S}
+        install -d  ${D}${nonarch_base_libdir}/firmware/
+        install -d  ${D}${nonarch_base_libdir}/firmware/qcom/sa8155p
+
+        install -m 0444 adsp*.* ${D}${nonarch_base_libdir}/firmware/qcom/sa8155p
+        install -m 0444 cdsp*.* ${D}${nonarch_base_libdir}/firmware/qcom/sa8155p
```
[the remark this answers is not in the document]
Comment (bhupesh-sharma) [reply recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb:28]: We have .mdt files files only - for cDSP and aDSP. No plan to pil-squach the .mdt file(s) into .mbn ones for the initial sa8155p support.

File: recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+DESCRIPTION = "QCOM Firmware for SA8155p ADP platform"
+
+# Provide base URI of 'sa8155p_adp_fw.zip' files here.
```
[the remark this answers is not in the document]
Comment (bhupesh-sharma) [reply recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb:3]: Archive is created sa8155p_adp_fw folder provided by QCoM using:

$ zip -r sa8155p_adp_fw.zip sa8155p_adp_fw

File: recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb (hunk 1 above)
Comment (bhupesh-sharma) [reply recipes-bsp/firmware/firmware-qcom-sa8155p_1.0.bb:38]: Ok.

**Pull request closed without merging**